lib/events.js

```diff
@@ -437,9 +441,9 @@ EventEmitter.prototype.removeAllListeners =
         this.removeListener(type, listeners);
       } else if (listeners) {
         // LIFO order
-        do {
-          this.removeListener(type, listeners[listeners.length - 1]);
-        } while (listeners[0]);
+        for (i = listeners.length; i-- > 0;) {
```

Comment:
I agree, this pattern is nigh unreadable. Let's change it in both places to something simpler.